src/Database/LSMTree/Internal/Run.hs

```diff
@@ -156,18 +156,36 @@ readBlob fs Run {..} BlobSpan {..} = do
 -- TODO: Once snapshots are implemented, files should get removed, but for now
 -- we want to be able to re-open closed runs from disk.
 close :: HasFS IO h -> HasBlockIO IO h -> Run (FS.Handle h) -> IO ()
-close fs _hbio Run {..} = do
+close fs hbio Run {..} = do
     FS.hClose fs runKOpsFile
       `finally`
         FS.hClose fs runBlobFile
+    -- TODO: removing files should drop them from the page cache, but until we
+    -- have proper snapshotting we are keeping the files around. Because of
+    -- this, we instruct the OS to drop all run-related files from the page
+    -- cache
+    mapM_ (FS.hDropCacheAll' fs hbio) (pathsForRunFiles runRunFsPaths)
 
 -- | Should this run cache key\/ops data in memory?
 data RunDataCaching = CacheRunData | NoCacheRunData
   deriving stock Eq
 
+instance NFData RunDataCaching where
+  rnf CacheRunData   = ()
+  rnf NoCacheRunData = ()
+
+isNoCacheRunData :: RunDataCaching -> Bool
+isNoCacheRunData CacheRunData   = False
+isNoCacheRunData NoCacheRunData = True
```

Comment:
Where this is used, I think it would be clearer to just use an equality test, e.g.
instead of
```
when (isNoCacheRunData runRunDataCaching) $ do
```
use
```
when (runRunDataCaching == NoCacheRunData) $ do
```
So I suggest dropping this helper function.